packages/gatsby-recipes/src/renderer/render.js

```diff
@@ -202,43 +214,86 @@ const handleResource = (resourceName, context, props) => {
 
   inFlightCache.set(cacheKey, promise)
 
-  queue.add(() => promise)
+  queue.push(promise)
 
   throw promise
 }
 
-const render = async (recipe, cb, inputs = {}, isApply) => {
+const render = (recipe, cb, inputs = {}, isApply, isStream, name) => {
+  const emitter = mitt()
   const plan = {}
 
+  const queue = new Queue(
+    async (job, cb) => {
+      const result = await job
+      cb(null, result)
+    },
+    { concurrent: 5 }
+  )
+
+  queue.push(new Promise(resolve => resolve()))
```

Comment:
Not sure I understand pushing this empty promise to the queue. Is this to ensure `isDrained` isn't true until there's an opportunity to begin the render which will queue up resource promises?

Reply:
ugh :-( this is testing code left over — I'll remove this + add the compare function